examples/forward.hlb

```diff
@@ -0,0 +1,47 @@
+fs default() {
+	image "alpine" with option { resolve; }
+	run "docker version" with option {
+		mount dockerCli "/usr/local/bin"
+		forward "unix:///run/docker.sock" "/run/docker.sock"
+	}
+}
+
+fs buildDockerCli() {
+	image "golang:alpine" with option { resolve; }
+	run "apk add -U git bash coreutils gcc musl-dev"
+	env "CGO_ENABLED" "0"
+	env "DISABLE_WARN_OUTSIDE_CONTAINER" "1"
+	run "./scripts/build/binary" with option {
+		dir "/go/src/github.com/docker/cli"
+		mount fs { git "https://github.com/docker/cli.git" "v19.03.8"; } "/go/src/github.com/docker/cli"
+		mount fs { scratch; } "/go/src/github.com/docker/cli/build" as dockerCli
+	}
+}
+
+fs ssh() {
+	image "alpine"
+	run "apk add -U openssh-client"
+	mkdir "/root/.ssh" 0o700
+	run "ssh-keyscan github.com >> /root/.ssh/known_hosts"
+	run "ssh -q -T git@github.com || true" with option {
+		ignoreCache
+		ssh
+	}
+}
+
+fs nginx() {
```

Comment:
So I understand, the nginx target is running an nginx container on my localhost via shared docker.sock file?  If we run this target twice, it will be cached (but the daemon might not necessarily be running locally, perhaps I rebooted my laptop) and not restart it?
Not sure we have a strategy around that.  `ignoreCache` would make sure it is run again, but then if the container is already running locally that will also cause a failure.

I think the example is interesting, but it seems like there are some "sharp edges" here.

Reply:
Yes, we can definitely improve the actual command there so it will have smoother edges. Perhaps something like `docker rm -f nginx || true`.